storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/BlindedHotBlockMigration.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.server.kvstore;
+
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.storage.server.kvstore.dataaccess.KvStoreCombinedDaoBlinded;
+import tech.pegasys.teku.storage.server.kvstore.dataaccess.KvStoreCombinedDaoUnblinded;
+
+public class BlindedHotBlockMigration<
+    T extends KvStoreCombinedDaoBlinded & KvStoreCombinedDaoUnblinded> {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final int BATCH_SIZE = 1_000;
+  private final Spec spec;
+
+  private final T dao;
+
+  private BlindedHotBlockMigration(final Spec spec, final T dao) {
+    this.spec = spec;
+    this.dao = dao;
+  }
+
+  static <T extends KvStoreCombinedDaoBlinded & KvStoreCombinedDaoUnblinded> void migrateBlocks(
+      final T dao, final Spec spec) {
+    final BlindedHotBlockMigration<T> blindedHotBlockMigration =
+        new BlindedHotBlockMigration<>(spec, dao);
+    blindedHotBlockMigration.performBatchMigration();
+  }
+
+  private void performBatchMigration() {
+    moveHotBlocksToBlindedStorage();
+  }
+
+  private void moveFirstFinalizedBlockToBlindedStorage() {
```

Comment:
very small nit: I'd move this method to after `moveHotBlocksToBlindedStorage` so the order of code in the file matches the order of execution.